### PR TITLE
Make all struct fields embedded or defined inline inline

### PR DESCRIPTION
**Pull request by erizocosmico**

### CODE REVIEW

File: generator/processor.go [hunk 1]
```diff
@@ -222,9 +222,17 @@ func (p *Processor) processFields(s *types.Struct, done []*types.Struct, root bo
...
 		}
 
+		var tname string
+		switch typ := f.Type().Underlying().(type) {
+		case *types.Struct:
+			tname = typeName(f.Type())
+		default:
+			tname = typeName(typ)
+		}
+
```
Comment (Serabe) [inline generator/processor.go]: I would extract this snippet to a `func` of its own.

**Comment by erizocosmico** [comment]
@dpordomingo yes, this case has to be removed

File: generator/processor.go (hunk 1 above)
Comment (erizocosmico) [inline generator/processor.go]: this has to be removed, don't worry

**Comment by erizocosmico** [comment]
Now it can be reviewed again @Serabe @dpordomingo

**Pull request merged**